[RUM][Mobile] RUM-11586: Documentation update for SDK v3. (#31622)
DataDog/documentation: layouts/shortcodes/android-otel-note.en.md

> The Datadog Tracer implements the [OpenTelemetry][1] standard, and we recommend using it as an interface for tracing your application because it's vendor-neutral, supports many languages and frameworks, and unifies traces, metrics, and logs under one standard. See [this][3] page for instructions on setting up OpenTelemetry integration with the SDK. However, the OpenTelemetry specification library **[requires desugaring][2]** to be enabled for projects with a `minSdk` < `26`. If this requirement cannot be met in your project, you can still use the Trace product with the Datadog API.

Reviewer: ```suggestion
The Datadog Tracer implements the [OpenTelemetry][1] standard, and Datadog recommends using it as an interface for tracing your application because it's vendor-neutral, supports many languages and frameworks, and unifies traces, metrics, and logs under one standard. See instructions on [setting up OpenTelemetry integration with the SDK][3]. 

**Note:** The OpenTelemetry specification library requires [desugaring][2] to be enabled for projects with a `minSdk` < `26`. If you cannot enable desugaring in your project, you can still use the Trace product with the Datadog API instead.
```

> **Note**: The *Datadog API* implementation helps you transition from OpenTracing to OpenTelemetry. However, it may be removed in future releases of the SDK.

Reviewer: it is our policy to document the product as it works now, without mentioning possible changes in the future :)

```suggestion
**Note**: The *Datadog API* implementation helps you transition from OpenTracing to OpenTelemetry.
```
